chromium_src/chrome/browser/lifetime/application_lifetime_desktop.cc

```diff
@@ -0,0 +1,25 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "chrome/browser/lifetime/application_lifetime_desktop.h"
+
+#include "brave/browser/ui/webui/settings/brave_relaunch_handler_mac.h"
+
+#define AttemptRestart AttemptRestart_ChromiumImpl
+#include "src/chrome/browser/lifetime/application_lifetime_desktop.cc"
+#undef AttemptRestart
+
+namespace chrome {
+
+void AttemptRestart() {
+#if BUILDFLAG(IS_MAC)
+  if (brave_relaunch_handler::RelaunchOnMac()) {
```

Comment:
Curious this change can cause infinite loop something like `AttemptRestart() from chrome` -> `handling relaunch from sparkle` -> `Relaunch request from sparkle`?
I also tried to fix this issue long times ago(https://github.com/brave/brave-core/pull/15257) and I experienced this loop when I try to use similar approach with this PR.

Reply:
It never caused such a loop in my tests.

Reply:
(My tests consisted of changing a setting on `brave://settings/privacy`, then clicking the associated _Relaunch_ button, at various states of an update in progress on `brave://settings/help`.)

Reply:
Ok, great if it didn't happen.